src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,291 @@
+        """
+
+        connection_id = connection.GetSettings()['connection']['id']
+        if connection_id == self.wired_network_name:
+            network_device = self.wired_device
+
+        elif connection_id in self.wifi_network_names:
+            network_device = self.wireless_device
+
+        else:
+            raise UnknownConnectionIDException('The connection id %s is not configured.' %
+                                               connection_id)
+
+        return network_device
+
+    def _get_connection_state(self, connection_id):
+        """Returns the state of the connection with the given network id.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+        state = NM_CONNECTION_DISCONNECTED
+
+        active_connection = self._get_active_connection(connection_id)
+
+        if active_connection is None:
+            self.logger.warning('Connection %s disconnected.', connection_id)
+
+        else:
+            if hasattr(active_connection, 'State'):
+                if active_connection.State \
+                        is NetworkManager.NM_ACTIVE_CONNECTION_STATE_ACTIVATING:
+                    state = NM_CONNECTION_ACTIVATING
+                if active_connection.State \
+                        is NetworkManager.NM_ACTIVE_CONNECTION_STATE_ACTIVATED:
+                    state = NM_CONNECTION_ACTIVATED
+
+            else:
+                self.logger.error('Active connection for connection %s is no longer valid.',
+                                  connection_id)
+
+        return state
+
+    def _get_active_connection(self, connection_id):
+        """Returns the active connection object associated with the given network id.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+        active_connection = None
+
+        active_connections = NetworkManager.NetworkManager.ActiveConnections
+        self._run_proxy_call(active_connections)
+
+        for available_active_connection in active_connections:
+            if available_active_connection.Id == connection_id:
+                self.logger.debug('Found active connection for connection %s.',
+                                  connection_id)
+                active_connection = available_active_connection
+
+        return active_connection
+
+    def _wait_for_connection(self, connection):
+        """Wait for the configured number of seconds for an active connection to be ready.
+        return True if it connects in that time, False otherwise.
+
+        connection: The NetworkManager.Connection object to watch for connectivity.
+        """
+        success = False
+        give_up = False
+        connection_id = connection.GetSettings()['connection']['id']
+
+        self.logger.debug('Waiting for connection %s...', connection_id)
+        time_to_give_up = time.time() + self.network_activation_timeout
```

Comment:
I like where you declared time_to_give_up before this pull request. (A few lines up.)